tests/unittests/lib.files.art

```diff
@@ -0,0 +1,934 @@
+    ensure -> every? folders 'f
+    -> exists?.directory ~"|dir|/dest/|f|"
+
+    print "empty folders moved to dest/"
+
+]
+
+
+topic "move.directory - with filled folders"
+do [
+
+    dir: createTestFolder "move-filled-dir"
+
+    folders: @["A" "B" "C"]
+    loop folders 'f -> write.directory ~"|dir|/input/|f|" null
+    loop folders 'f -> write ~"|dir|/input/|f|/file.txt" "."
+    write.directory ~"|dir|/dest" null
+
+    ensure -> every? folders 'f -> exists? ~"|dir|/input/|f|/file.txt"
+
+    ; ---------//---------//---------//
+
+    move.directory ~"|dir|/input" ~"|dir|/dest"
+
+    ensure -> not? exists?.directory ~"|dir|/input"
+    ensure -> every? folders 'f
+    -> exists? ~"|dir|/dest/|f|/file.txt"
+
+    print "filled folders moved to dest/"
+
+]
+
+
+; -- Test: permissions
+
+topic "permissions"
+do [
+
+    dir: createTestFolder "permissions"
+
+    write ~"|dir|/file.txt" "Hello, world!"
+    ensure -> exists? ~"|dir|/file.txt"
+
+    ; ---------//---------//---------//
+    inspect permissions ~"|dir|/file.txt"
+
+]
+
+
+;; Returns me a `SIGSEGV: Illegal storage access. (Attempt to read from nil?)`
+; topic "permissions.set"
+; do [
+;
+;     dir: createTestFolder "permissions-set"
+;
+;     write ~"|dir|/file.txt" "Hello, world!"
+;
+;     ; action
+;     permissions.set: #[others:#[write:false]] ~"|dir|/file.txt"
+;
+;     ; assertion
+;     inspect permissions ~"|dir|/file.txt"
+;
+;
+; ]
+
+; -- Test: read
+
+topic "read"
+do [
+
+    dir: createTestFolder "read"
+
+    write ~"|dir|/test.txt" {
+        Hello, world
+        This is a multiline File.
+        :)
+    }
+    ensure -> exists? ~"|dir|/test.txt"
+
+    ; ---------//---------//---------//
+
+    inspect read ~"|dir|/test.txt"
+
+]
+
+
+topic "read.lines"
+do [
+
+    dir: createTestFolder "read-lines"
+
+    write ~"|dir|/test.txt" {
+        The Language
+        ------------------------------
+
+        Arturo is an independently-developed, modern programming language,
+        vaguely related to various other ones - including but not limited to:
+        Logo, Rebol, Forth, Ruby, Haskell, D, Smalltalk, Tcl, and Lisp.
+
+        The language has been designed
+        following some very simple and straightforward principles:
+
+        - Code is just a list of words, symbols and literal values
+        - Words and symbols within a block are interpreted - when needed -
+            according to the context
+        - No reserved words or keywords -
+            look for them as hard as you can; there are absolutely none
+    }
+
+    ensure -> exists? ~"|dir|/test.txt"
+
+    ; ---------//---------//---------//
+
+    inspect read.lines ~"|dir|/test.txt"
+
+]
+
+topic "read.json"
+do [
+
+    dir: createTestFolder "read-json"
+
+    write ~"|dir|/test.json" {
+        {
+            "name": "Arturo",
+            "version": "0.9.83",
+            "build": "b/12",
+            "platform": "amd/win10"
+        }
+    }
```

Comment:
```suggestion
    jsonSample: {
        {
            "name": "Arturo",
            "version": "0.9.83",
            "build": "b/12",
            "platform": "amd/win10"
        }
    }
    write ~"|dir|/test.json" jsonSample
    
    inspect read.json jsonSample
```

Also - not that it's *that* important, but since you've done it all in such a nice and organized way - since `read` (or `read.json`, `read.html`, etc for that matter) also work directly on string values, you could also directly test the string value and... then when it's reading from a file, as you already do. 😉 

Just an additional idea - but I'd say we'd better manage to get it to pass the tests first, as it is.

Reply:
Alright! 😉

Reply:
Replaced! at [97f911c](https://github.com/arturo-lang/arturo/pull/1008/commits/97f911c709d7714ddad95ff725af1803f69cde68)